### PR TITLE
Replace dummy benchmark

### PR DESCRIPTION
**Pull request by vext01**
Don't merge yet, outstanding issue below.

This:
 * replaces the dummy benchmark with fannkuch_redux
 * removes a lot of duplication between the various example configs
 * fixes Java support build instructions

One outstanding issue I'd like to understand:

```
[2017-10-10 11:05:05: ERROR] Benchmark returned non-zero or emitted invalid JSON.
Exception string: No JSON object could be decoded    
return code: 0    
stdout:    
--------------------------------------------------    
    
--------------------------------------------------    
    
stderr:    
--------------------------------------------------    
iterations_runner_c: /home/edd/research/krun/examples/benchmarks/fannkuch_redux/c/bench.so: undefined symbol: rotate
    
--------------------------------------------------    
```

Yet, here is the symbol:
https://github.com/softdevteam/krun/blob/71975a53baff85034a37888b617b092c227351e9/examples/benchmarks/fannkuch_redux/c/bench.c#L45-L52

If I remove `inline` from the signature, then all is well. What gives?

### CODE REVIEW

**Comment by ltratt** [comment]
If you force `rotate` to be inlined, then presumably that function doesn't actually exist in the binary.

[Personally I think `inline` in modern C is a bad idea: the compiler generally does a better job than humans of guessing when to use it. So I'd just drop it.]

**Comment by snim2** [comment]
There are some `.env` files committed here, was that intentional?

**Comment by vext01** [comment]
Alright. A PR will follow to do the same on the warmup_experiment repo.

Will fix tests.

**Comment by vext01** [comment]
@snim2 no, but i killed them in a later commit

**Comment by snim2** [comment]
ok, cool

**Comment by vext01** [comment]
Ready for re-review.

Note the doc fix too.

**Comment by vext01** [comment]
Fixes #354

File: examples/benchmarks/fannkuch_redux/java/KrunEntry.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+class KrunEntry implements BaseKrunEntry {
+  static { fannkuchredux.init(); } // force class to be loaded
```
Comment (ltratt) [inline examples/benchmarks/fannkuch_redux/java/KrunEntry.java]: Weird mix of indentation?

File: examples/benchmarks/fannkuch_redux/java/fannkuchredux.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+*/
+
+public class fannkuchredux
+{
+    static void init() {};
+    private static int EXPECT_CKSUM = 1616;
+    private static int MAX_N = 8;
+
+
+   public static int inner_iter() {
```
Comment (vext01) [inline examples/benchmarks/fannkuch_redux/java/fannkuchredux.java]: and here

Comment (ltratt) [reply examples/benchmarks/fannkuch_redux/java/fannkuchredux.java]: Good spot.

File: examples/benchmarks/fannkuch_redux/java/fannkuchredux.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            count[r] = count[r] - 1;
+            if (count[r] > 0) break;
+            r++;
+         }
+
+         permCount++;
+      }
+   }
+
+   public static void runIter(int n) {
```
Comment (vext01) [inline examples/benchmarks/fannkuch_redux/java/fannkuchredux.java]: and here

**Comment by vext01** [comment]
Hrm, shall I re-indent the whole file?

**Comment by ltratt** [comment]
Probably a good idea.

**Comment by vext01** [comment]
There we go.

So can I squash this?

**Comment by ltratt** [comment]
Yes, please squash.

**Comment by vext01** [comment]
Now that's some quality squashing right there :)

**Comment by vext01** [comment]
Merge when ready.

**Pull request merged**